pkg/build/controller/strategy/util.go

```diff
@@ -93,7 +94,11 @@ func setupDockerSocket(pod *v1.Pod) {
 // mountSecretVolume is a helper method responsible for actual mounting secret
 // volumes into a pod.
 func mountSecretVolume(pod *v1.Pod, container *v1.Container, secretName, mountPath, volumeSuffix string) {
-	volumeName := namer.GetName(secretName, volumeSuffix, kvalidation.DNS1123SubdomainMaxLength)
+	volumeName := namer.GetName(secretName, volumeSuffix, kvalidation.DNS1123LabelMaxLength)
+
+	// coerce from RFC1123 subdomain to RFC1123 label.
+	volumeName = strings.Replace(volumeName, ".", "-", -1)
```

Comment:
not particularly.  if you really have two secrets like that, we have no way to support it anyway and the pod creation will fail when we try to define two volumes w/ the same name (I would assume).  You're stuck regardless.

if we were really worried about it we could try multiple de-duplication replacements, but someone having "my.secret" and "my-secret" and wanting them both in a single build isn't terribly high on my list of worries, especially since this is the first we've even heard of someone trying to use one with a dot in the name.